### ISSUE TITLE
Use Alien::hiredis?

### ISSUE THREAD

**Issue by Grinnz** [neophenix/Redis-hiredis#11]
I have a distribution called [Alien::hiredis](https://metacpan.org/pod/Alien::hiredis) which could be used by this module. See [Alien](https://metacpan.org/pod/Alien) for an overview of the Alien ecosystem, but the basic advantages are that the task of building and updating the hiredis library is outsourced to that module, and it can even allow you to build against a system version of hiredis if it is appropriate.

All you would need to do is use it like described [here](https://metacpan.org/pod/Alien::Build::Manual::AlienUser#ExtUtils::MakeMaker) (I can provide a PR if you wish). Right now it requires hiredis 0.11.0 or newer, with an exception to disallow a system version of 0.13.0 which is a broken version. In other cases it will install a custom version of the library for Perl to use. This is configured by the [alienfile](https://metacpan.org/source/DBOOK/Alien-hiredis-0.008/alienfile).

You would not be able to affect what criteria it uses to determine whether to use the system version or install a new one, but it could be adjusted if needed, and you can conditionally fail to install based on the [alien library version](https://metacpan.org/pod/Alien::Base#atleast_version).

Just presenting this option in case you want to let something else worry about installing or updating the library, and allow usage of system hiredis. Thanks for your consideration.

**Comment by neophenix** [comment]
Hey thanks for the idea.  I think doing that would make this module largely pointless then as its already a basic wrapper over hiredis, I don't think adding another layer there would make a lot of sense for what my module tries to accomplish.

The building, etc of hiredis was always pretty simple, right now its just the fact I no longer use Perl or Redis day to day so its finding time and motivation to do any updates.

**Comment by Grinnz** [comment]
I don't think it would make this module pointless at all, since the Alien does not provide any Perl interface to the library, but instead an interface for XS (like this module) or FFI to use it. But no pressure to use it unless you think it would be beneficial.
